src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -914,21 +914,29 @@ private void ParseHelper(string[] args)
                 else if (MatchSwitch(switchKey, "socketservermode", "so"))
                 {
                     _socketServerMode = true;
+                    _showBanner = false;
+                    _noProfileLoadTime = true;
```

Comment:
For completeness, I prefer the existing explicit setting of each switch.

Reply:
I looked a bit more on where the profile gets evaluated and the message about slow profile execution gets reported, and it turns out that code path never gets hit when the any of the server modes is enabled.

See the code pointed by
https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs#L201-L273

The evaluation of profile is in [DoRunspaceInitialization](https://github.com/PowerShell/PowerShell/blob/4051e097965019fdd681a90eafd63e63aa5219f1/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs#L1777), which is called only from `s_theConsoleHost.Run(s_cpp, false);` from the final `else` block in the above code. Therefore, there is no need to set `_noProfileLoadTime` here, and setting it could be misleading.

Reply:
I am fine with removing it.  My only concern is if the code is refactored in the future and the switch does become effective.  I like explicitly setting it just to be safe.  But I am good either way.